plugins/osdn/registry.go

```diff
@@ -145,12 +149,14 @@ func (registry *Registry) WatchPods() error {
 		}
 		pod := obj.(*kapi.Pod)
 
+		registry.podTrackingLock.Lock()
 		switch eventType {
 		case watch.Added, watch.Modified:
 			registry.trackPod(pod)
 		case watch.Deleted:
 			registry.unTrackPod(pod)
 		}
+		registry.podTrackingLock.Unlock()
```

Comment:
> Instead I prefer adding these 3 methods:
> ...
> Use these methods where ever required : in endpointsUpdate, trackPod and untrackPod

The lock needs to be held throughout trackPod() and untrackPod() though or else you might test one value and then delete another, etc.

Re-pushed with proper go-ish use of defer to unlock the mutexes, and changed trackPod() to delete stale data itself rather than calling out to unTrackPod() to do it. (unTrackPod() does an additional check which is dropped now, but that check should be irrelevant; if the UID is correct, then we want to remove the pod, regardless of whether the IP matches or not [which it should, but...])